apps/dav/lib/CalDAV/EventReaderRRule.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright 2024 Sebastian Krupinski <krupinski01@gmail.com>
+ *
+ * @author Sebastian Krupinski <krupinski01@gmail.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+namespace OCA\DAV\CalDAV;
+
+class EventReaderRRule extends \Sabre\VObject\Recur\RRuleIterator {
+
+	public function precision(): string {
+		return $this->frequency;
+	}
+
+	public function interval(): int {
+		return $this->interval;
+	}
+
+	public function concludes(): \DateTime | null {
+		if (isset($this->until)) {
```

Comment:
```suggestion
		if ($this->until instanceof \DateTimeInterface) {
```

isset is totally fine because until is typed as \DateTimeInterface|null. 
I prefer the instanceof version for the better readability.